rewrite-java/src/main/java/org/openrewrite/java/format/BlankLinesProcessor.java

```diff
@@ -46,16 +46,13 @@ public BlankLinesProcessor(BlankLineStyle style) {
                     return c.withSuffix(suffix);
                 }));
             } else {
-                j = minimumLines(j, style.getMinimum().getBeforePackage());
+                j = j.withPrefix(j.getPrefix().withWhitespace(""));
```

Comment:
The package exists. We need to ensure that it has a minimum number of lines according to the `beforePackage` setting. Setting it to blank doesn't seem right.

Reply:
I added a comment to make this a bit more clear, but this turns out to be due to a case where when comments are empty/missing and the package decl is present, a leading newline on the compilation unit gets removed by intellij formatting. There was a unit test to this effect that was failing.